### PR TITLE
Producing meta modules for packages like wrf and roms

### PR DESCRIPTION
**Pull request by pelahi**
This PR is currently a draft as I am testing ideas on generating modules that load all the dependencies need by a package.

### CODE REVIEW

**Comment by marcodelapierre** [comment]
Thanks Pascal, I will review as soon as I am back to work.

One thing that I wanted to share with you on the wrf/roms meta-modules, which is actually just a heads up, because I think your current implementation is fine with that.

The strings `wrf-dependencies` and `roms-dependencies` may negatively interact with the string filtering that happens in the module template:
https://github.com/PawseySC/pawsey-spack-config/blob/main/setonix/templates_setonix/modules/modulefile.lua#L73

However, yout current implementation is OK, because you have a `.lua` extension, `wrf-dependencies.lua`, whereas the filter strips out `dependencies/`, with a final slash.

**Comment by pelahi** [comment]
Thanks @marcodelapierre  for the point. I had thought of that and wanted to avoid `dependencies` but that is what the module provides.

**Comment by marcodelapierre** [comment]
Oh, one extra though on this, sorry my brain is slower than usual today. It might get tricky in case we need to change the naming convention for these modules in the future, e.g. from `<tool>-dependencies.lua` to `<tool>-dependencies/<version>/module.lua`, to account for versions. then, the filter will play dirty tricks.

Then, one alternative I was thinking of is `<tool>-dependency-set`, because in general it will provide a set of dependency.

**Comment by pelahi** [comment]
So now the module produced is something like `wrf-dependency-set/<version>/module.lua` 
```lua
-- -*- lua -*-
-- Module file created by Pawsey
--

whatis([[Name : wrf-dendencies]])
whatis([[Package version and spec used to generate deps:   wrf@4.2%gcc@10.3.0 cflags="-O3" cppflags="-O3" fflags="-O3 -fallow-argument-mismatch" +pnetcd
f build_type=dmpar compile_type=em_real nesting=basic patches=01c0f71a4c742603ca6418977bdc71c8c5ccf882e54e6c193ec2adaf98e31511,0b093750c1344ecb15061ad23
634055a39cc79698a396bf6f6c015a6c856c823,68548f61600d8503bf14a1ab8b90e0e20375fe9466fa1e205b8d835174fcdbef,b3f063c60c61ac6655f91cbb266e3a0dafc06449ebb701c
b56541c5e9bff65e9,b3fc8bb1e6924f9441e746d0de7dc2b2421b064272a6cc33e8cf8ce8c1ad20a9,c47ec9cba9b7396186801f9eaae794fe6db53570486095d1bda4f58e327b4eec,d019
b9605304d5728d6eceda24b1ef4398227fd978f3c1b32a91fbc4f5069e8d,e07c39c644dab0cad373e4ff3809db22128b376c139416d9e774b96e1f86e7e7,f3dd50d3de8dc26d12ab5bceea
a80ec05a722834dde97107d7bb194c9f67045a,f85f29a41d535b1ca612bcf8217a3ea4975fb0b95dcd69e2bf0b26605795261c arch=cray-sles15-zen2 ]])

whatis([[Short description : Meta module to load dependencies to build wrf. ]])
help([[Meta module that loads all dependencies need to build local wrf]])

-- TODO: figure out way of spack generating dependencies
-- currently would be to make use of the spack.lock
-- in the related environment.
load("dependencies/.cray-mpich/7.7.0-rpq4ivk/module")
load("libraries/hdf5/1.12.1-parallel-api-v112/module")
load("utilities/cmake/3.21.4/module")
load("dependencies/.ncurses/6.1-f5xbinw/module")
load("dependencies/.openssl/1.1.0i-fips-kl6ezjb/module")
load("dependencies/.libaec/1.0.5-wa6rg3n/module")
load("dependencies/.numactl/2.0.11-tinrgvs/module")
load("dependencies/.pkg-config/0.29.2-3tlf6jx/module")
load("dependencies/.zlib/1.2.11-fhca6le/module")
load("dependencies/.jasper/2.0.32-dfek24i/module")
load("dependencies/.libjpeg-turbo/2.1.0-aygwfi2/module")
load("dependencies/.nasm/2.15.05-4ooqntw/module")
load("dependencies/.libpng/1.6.37-6cux2z5/module")
load("dependencies/.libtirpc/1.2.6-tl666wp/module")
load("dependencies/.krb5/1.19.2-5lln5yk/module")
load("dependencies/.libtool/2.4.6-jlvb3j2/module")
load("libraries/netcdf-c/4.8.1/module")
load("dependencies/.hdf/4.2.15-2zux5tl/module")
load("libraries/parallel-netcdf/1.12.2/module")
load("programming-languages/perl/5.34.0/module")
load("dependencies/.berkeley-db/18.1.40-ihkuhwv/module")
load("dependencies/.bzip2/1.0.6-t34oqcl/module")
load("dependencies/.gdbm/1.19-72vdi62/module")
load("dependencies/.readline/8.1-pdzbirz/module")
load("libraries/netcdf-fortran/4.5.3/module")
load("dependencies/.tcsh/6.22.02-d7xyudn/module")
```

Just now need to strip the key words from the lua file.

**Comment by pelahi** [comment]
Now 
```lua
-- -*- lua -*-
-- Module file created by Pawsey
--

whatis([[Name : wrf-dependency-set]])
whatis([[Package version and spec used to generate deps:   wrf@4.2%gcc@10.3.0 cflags="-O3" cppflags="-O3" fflags="-O3 -fallow-argument-mismatch" +pnetcd
f build_type=dmpar compile_type=em_real nesting=basic patches=01c0f71a4c742603ca6418977bdc71c8c5ccf882e54e6c193ec2adaf98e31511,0b093750c1344ecb15061ad23
634055a39cc79698a396bf6f6c015a6c856c823,68548f61600d8503bf14a1ab8b90e0e20375fe9466fa1e205b8d835174fcdbef,b3f063c60c61ac6655f91cbb266e3a0dafc06449ebb701c
b56541c5e9bff65e9,b3fc8bb1e6924f9441e746d0de7dc2b2421b064272a6cc33e8cf8ce8c1ad20a9,c47ec9cba9b7396186801f9eaae794fe6db53570486095d1bda4f58e327b4eec,d019
b9605304d5728d6eceda24b1ef4398227fd978f3c1b32a91fbc4f5069e8d,e07c39c644dab0cad373e4ff3809db22128b376c139416d9e774b96e1f86e7e7,f3dd50d3de8dc26d12ab5bceea
a80ec05a722834dde97107d7bb194c9f67045a,f85f29a41d535b1ca612bcf8217a3ea4975fb0b95dcd69e2bf0b26605795261c arch=cray-sles15-zen2 ]])

whatis([[Short description : Meta module to load dependencies to build wrf@4.2. ]])
help([[Meta module that loads all dependencies need to build local wrf@4.2]])

-- TODO: figure out way of spack generating dependencies
-- currently would be to make use of the spack.lock
-- in the related environment.
load(.cray-mpich/7.7.0-rpq4ivk/module")
load(hdf5/1.12.1-parallel-api-v112/module")
load(cmake/3.21.4/module")
load(.ncurses/6.1-f5xbinw/module")
load(.openssl/1.1.0i-fips-kl6ezjb/module")
load(.libaec/1.0.5-wa6rg3n/module")
load(.numactl/2.0.11-tinrgvs/module")
load(.pkg-config/0.29.2-3tlf6jx/module")
load(.zlib/1.2.11-fhca6le/module")
load(.jasper/2.0.32-dfek24i/module")
load(.libjpeg-turbo/2.1.0-aygwfi2/module")
load(.nasm/2.15.05-4ooqntw/module")
load(.libpng/1.6.37-6cux2z5/module")
load(.libtirpc/1.2.6-tl666wp/module")
load(.krb5/1.19.2-5lln5yk/module")
load(.libtool/2.4.6-jlvb3j2/module")
load(netcdf-c/4.8.1/module")
load(.hdf/4.2.15-2zux5tl/module")
load(parallel-netcdf/1.12.2/module")
load(perl/5.34.0/module")
load(.berkeley-db/18.1.40-ihkuhwv/module")
load(.bzip2/1.0.6-t34oqcl/module")
load(.gdbm/1.19-72vdi62/module")
```

File: setonix/setup/make_metamoduledependencies.sh [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+if [ "$#" -ne 3 ];  then
+    echo "Usage is <package> <path_to_env> <path_to_modules>"
+    echo "Script will cd to the environment, activate it, check the concretisation, update the module and copy it to the module path"
+    exit
+fi
+
+package=$1
+envpath=$2
+modpath=$3
+script_dir="$( cd -- "$(dirname "$0")" >/dev/null 2>&1 ; pwd -P )"
```
Comment (marcodelapierre) [inline setonix/setup/make_metamoduledependencies.sh:13]: this might do it , too (to be checked):
```
script_dir="$( readlink -f "$(dirname "$0")" )"
```

**Review by marcodelapierre — APPROVED** [review]

**Comment by marcodelapierre** [comment]
thanks @pelahi , very important piece of work for the Setonix Spack stack

**Pull request merged**